src/tests/payment/test_views.py

```diff
@@ -0,0 +1,176 @@
+from unittest import mock
+from unittest.mock import MagicMock, patch
+
+import pytest
+import stripe
+from django.urls.base import reverse_lazy
+from rest_framework import status
+
+from payment.models import StripeProduct, Subscription
+
+
+@pytest.fixture
+def client(authenticated_client):
+    return authenticated_client
+
+
+@pytest.mark.django_db
+class TestSubscriptionViewSet:
+
+    create_url = reverse_lazy("v1:payment:subscription-list")
+    cancel_user_url = reverse_lazy("v1:payment:subscription-user-cancel")
+
+    def test_endpoint(self):
+        assert self.create_url == "/api/v1/payment/subscription/"
+
+    def test_endpoint2(self):
+        assert self.cancel_user_url == "/api/v1/payment/subscriptions/user/cancel/"
+
+    def retrieve_url(self, pk):
+        return reverse_lazy("v1:payment:subscription-detail", kwargs={"pk": pk})
+
+    @patch("payment.views.stripe.checkout.Session.create")
+    @patch("payment.views.StripeProduct.objects.get")
+    def test_create_subscription_success(
+        self, mock_get_product, mock_create_session, client, user
+    ):
+        client.force_login(user)
+        data = {"plan": "monthly"}
+        mock_checkout_session = MagicMock()
+        mock_checkout_session.id = "session_id_example"
+        mock_checkout_session.url = "https://checkout.stripe.com/pay/example"
+        mock_create_session.return_value = mock_checkout_session
+        response = client.post(self.create_url, data, format="json")
+        assert response.status_code == status.HTTP_201_CREATED
+        assert response.data["sessionId"] == "session_id_example"
+        assert (
+            response.data["checkout_url"] == "https://checkout.stripe.com/pay/example"
+        )
+
+    @patch("payment.views.stripe.checkout.Session.create")
+    @patch("payment.views.StripeProduct.objects.get")
+    def test_create_subscription_product_not_found(
+        self, mock_get_product, mock_create_session, client, user
+    ):
+        client.force_login(user)
+        data = {"plan": "monthly"}
+        mock_get_product.side_effect = StripeProduct.DoesNotExist
+        response = client.post(self.create_url, data, format="json")
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+        assert response.data["error"] == "Product not found"
```

Comment:
Assert the full data object when it's trivial enough, same thing on the other assert